### PR TITLE
[risk=low][no ticket] Remove deprecated endpoints and related code

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

Remove 3 endpoints which are no longer referenced by the UI because they have been replaced by newer versions.

~One of these is still referenced by the Perf tests, so we should wait until that reference is removed ( https://github.com/broadinstitute/mcnulty/pull/266 ) and a Perf run completes successfully with the new version.~ **GOOD TO GO**.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/CdrVersionsController.java [hunk 1]
```diff
@@ -16,12 +15,6 @@
     this.cdrVersionService = cdrVersionService;
   }
 
-  @Override
-  @Deprecated
-  public ResponseEntity<CdrVersionListResponse> getCdrVersions() {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/CdrVersionsController.java:21]: UI now calls `getCdrVersionsByTier()` instead

File: api/src/main/java/org/pmiops/workbench/api/InstitutionController.java [hunk 2]
```diff
@@ -88,22 +88,6 @@ private Institution getInstitutionImpl(final String shortName) {
...
   }
 
-  /**
-   * Note: this API is publicly-accessible since it is called during account creation.
-   *
-   * @return Returns whether the email is a valid institutional member.
-   * @deprecated in favor of the newer POST checkEmail() endpoint
-   */
-  @Override
-  public ResponseEntity<CheckEmailResponse> deprecatedCheckEmail(
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/InstitutionController.java:98]: UI now calls `checkEmail()` instead

File: api/src/main/java/org/pmiops/workbench/api/UserController.java [hunk 3]
```diff
@@ -80,55 +80,6 @@ public UserController(
...
-   * @param pageToken
-   * @param pageSize
-   * @param sortOrder
-   * @return
-   * @deprecated use {@link #userSearch(String, String, String, Integer, String)} with an access
-   *     tier short name argument instead.
-   */
-  @Override
-  @Deprecated
-  public ResponseEntity<UserResponse> user(
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/UserController.java:96]: UI now calls `userSearch()` instead

File: api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java [hunk 4]
```diff
@@ -612,6 +612,9 @@ private FirecloudWorkspaceId getFcBillingProject(DbAccessTier accessTier, Worksp
   public ResponseEntity<EmptyResponse> reviewWorkspace(
       String ns, String id, ResearchPurposeReviewRequest review) {
     DbWorkspace workspace = workspaceDao.getRequired(ns, id);
+
+    // RW-7087 replace with a new workspaceAuditor action (fireReviewAction?)
+    // because this uses the deprecated DbAdminActionHistory
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java:617]: calling out the only remaining use for a future story

File: api/src/test/java/org/pmiops/workbench/cdr/CdrVersionServiceTest.java [hunk 5]
```diff
@@ -258,32 +257,6 @@ public void testSetCdrVersionControlledIdForbiddenNotInGroup() {
         });
   }
 
-  // Tests for deprecated registered-tier-only getCdrVersions()
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/cdr/CdrVersionServiceTest.java:261]: equivalent tests for the new version are below

**Review by calbach — APPROVED** [review]
Nice, thanks for the cleanup

**Pull request merged**